### PR TITLE
feat(unlock-app, unlock-js): better support for v10

### LINKED ISSUE #8629: Checkout doesn't show if a person has a valid key on v10 lock
**Describe the bug**
When a user with existing keys goes through v10 lock checkout, they won't get an indication that they already have a key and are able to unlock the page. This functionality is practically broken right now. 

**To Reproduce**
1. Create a v10 lock
2. Go to the demo page
3. Buy a key and wait for it to be minted.
4. Reload and go through the checkout again. 

**Expected behavior**
You should get to the *You have a valid key to access this page.* checkout stage with the ability to buy more keys with buy for someone else.

### PR DESCRIPTION
**Pull request by julien51**
# Description

The new version of PublicLock enables multiple purchases for users. 
This pull requests adds better support for this.

# Issues

Fixes #8629

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/src/__tests__/integration/walletServiceIntegration.test.js [hunk 1]
```diff
@@ -304,14 +304,13 @@ describe.each(UnlockVersionNumbers)('Unlock %s', (unlockVersion) => {
...
           )
-          key =
-            ['v10'].indexOf(publicLockVersion) !== -1
-              ? await web3Service.getKeyByTokenId(lockAddress, tokenId, chainId)
-              : await web3Service.getKeyByLockForOwner(
-                  lockAddress,
-                  keyGrantee,
-                  chainId
-                )
+          key = ['v11'].includes(publicLockVersion)
```
Comment (kalidiagne) [inline packages/unlock-js/src/__tests__/integration/walletServiceIntegration.test.js:307]: @julien51 changed everywhere we call `getKeyByTokenId ` from  `v10 -> v11` based this lines on `web3service.ts` 

```
async getKeyByTokenId(lockAddress: string, tokenId: string, network: number) {
    const lockContract = await this.getLockContract(
      lockAddress,
      this.providerForNetwork(network)
    )
    if ((await lockContract.publicLockVersion()) < 11) {
      throw new Error('Only available for Lock v11+')
    }
```

Comment (julien51) [reply packages/unlock-js/src/__tests__/integration/walletServiceIntegration.test.js:307]: No no we should try to keep a consistebt API.

Lets use web3Service.getKeyByLockForOwner which should work for all versions of public lock please.

Comment (julien51) [reply packages/unlock-js/src/__tests__/integration/walletServiceIntegration.test.js:307]: I added it intentionally to this PR. It was a mistake that it was missing

**Review by julien51 — COMMENTED** [review]
I think that works

**Review by kalidiagne — APPROVED** [review]

**Comment by julien51** [comment]
Wooooojoooo

**Pull request merged**